### ISSUE TITLE
LSP sometimes does not offer expected completions

### ISSUE THREAD

**Issue by abeldekat** [Saghen/blink.cmp#719]
### Make sure you have done the following

- [X] I have updated to the latest version of `blink.cmp`
- [X] I have read the README

### Bug Description

Given a simple rust test project, `cargo new atest`:

```rust
use std::vec;

fn main() {
    let y = Some(42);
    let _ = vec![42, 2];
}
```

I try to change text "42" in the `vec` on line 5 into `y.take().unwrap()`.
However, after typing `y.t` blink does not offer "take" as completion.

This was working in release `0.7.6`. I bisected `main` manually and found commit `f9b72e3c1a1b` to be the culprit. The description: "fix: run callback when LSP client returns nil"

Related: #543

### Relevant configuration

```lua
default blink
```

### neovim version

0.10.2

### `blink.cmp` version: branch, tag, or commit

main

**Comment by LeedsJohn** [comment]
Have you had any luck solving this problem?  I'm running into a similar issue, but it could very well be a problem with my config.

**Comment by abeldekat** [comment]
No, I think it can only be fixed by providing a PR. My knowledge of blink is not good enough to fix #543 in a different way.

[This](https://github.com/Saghen/blink.cmp/blob/0350feedfa8adb07b6750f6d9150c26e13eae0d2/lua/blink/cmp/sources/lsp.lua#L85) is the line that causes the problem.

**Comment by Saghen** [comment]
It's likely that https://github.com/Saghen/blink.cmp/commit/86a13aeb104d6ea782557518ee1a350712df7bd7 resolve this, lmk if you're still able to reproduce

**Comment by abeldekat** [comment]
I can still reproduce the issue. 

@LeedsJohn, could you also try?

I do notice something I have seen before but did not deem important.
In the example, when typing `y.` below line `let y = Some(42);` completion kicks in immediately on the `y` and refreshes after the `.`.

In the situation of the issue, completion kicks in when typing `y` but *stops* when typing `.`. I need to type another letter after the dot, after which the completion is not relevant anymore.

**Comment by Saghen** [comment]
30 minutes for a one line change :massage_man: serenity now

**Comment by LeedsJohn** [comment]
@abeldekat Sorry to be unhelpful, but when I installed blink.cmp with the feature that Saghen recommended, I got an error for an out of date or missing fuzzy matching library.  I briefly looked through the documentation for fuzzy configuration but wasn't able to fix it.  (I'm fairly confident that this is an issue on my end, and not a problem with blink.cmp).  If I have time later, I'll look at it again.

Saghen, thanks for working on this issue!
